policy/nomad/validate.go

```diff
@@ -172,17 +170,107 @@ func validateStrategy(s map[string]interface{}) error {
 	nameKey := "name"
 	nameInterface, ok := s[nameKey]
 	if !ok {
-		result = multierror.Append(result, fmt.Errorf(`%s is missing key "%s"`, path, nameKey))
+		result = multierror.Append(result, fmt.Errorf("%s->%s is missing", path, nameKey))
 	} else {
 		nameString, ok := nameInterface.(string)
 		if !ok {
-			result = multierror.Append(result, fmt.Errorf("%s[%s] must be string, found %T", path, nameKey, nameInterface))
+			result = multierror.Append(result, fmt.Errorf("%s->%s must be string, found %T", path, nameKey, nameInterface))
 		} else {
 			if nameString == "" {
-				result = multierror.Append(result, fmt.Errorf("%s[%s] can't be empty", path, nameKey))
+				result = multierror.Append(result, fmt.Errorf("%s->%s can't be empty", path, nameKey))
 			}
 		}
 	}
 
+	// Validate config (optional).
+	//   1. Config must be an HCL block if present.
+	configKey := "config"
+	if config, ok := s[configKey]; ok {
+		err := validateHCLBlock(config, path, configKey, nil)
+		if err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+
+	return result.ErrorOrNil()
+}
+
+// validateTarget validates the content of the target block inside policy.
+//
+//  scaling {
+//    policy {
+//      target = {
+//      +-----------------+
+//      | name = "target" |
+//      | config = {      |
+//      |   key = "value" |
+//      | }               |
+//      +-----------------+
+//      }
+//    }
+//  }
+func validateTarget(t map[string]interface{}) error {
+	var path = fmt.Sprintf("scaling->policy->%s", keyTarget)
+
+	var result *multierror.Error
+
+	// It shouldn't happen, but it's better to prevent a panic.
+	if t == nil {
+		return multierror.Append(result, fmt.Errorf("%s is nil", path))
+	}
+
+	// Validate name (optional).
+	//   1. Name must have string value if present.
+	//   2. Name must not be empty if present.
+	nameKey := "name"
+	nameInterface, ok := t[nameKey]
+	if ok {
+		nameString, ok := nameInterface.(string)
+		if !ok {
+			result = multierror.Append(result, fmt.Errorf("%s->%s must be string, found %T", path, nameKey, nameInterface))
+		} else {
+			if nameString == "" {
+				result = multierror.Append(result, fmt.Errorf("%s->%s can't be empty", path, nameKey))
+			}
+		}
+	}
+
+	// Validate config (optional).
+	//   1. Config must be an HCL block if present.
+	configKey := "config"
+	if config, ok := t[configKey]; ok {
+		err := validateHCLBlock(config, path, configKey, nil)
+		if err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+
+	return result.ErrorOrNil()
+}
+
+// validateHCLBlock validates the kind of unusual structure we receive when the policy HCL block is parsed.
+func validateHCLBlock(in interface{}, path, key string, validator func(in map[string]interface{}) error) error {
+	var result *multierror.Error
+
+	list, ok := in.([]interface{})
```

Comment:
is it worthwhile allowing this to short-circuit if it's already `map[string]interface{}`?

Reply:
Good idea, this also future-proof this code in case we change the Nomad API to return just the `map`.